rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -463,14 +464,21 @@ def _paginate(query, pagination, get_all_results=False, locking=False):
             size = config.instance.default_page_size
             offset = 0
 
-        total = query.order_by(None).count()  # Fastest way to count
         if locking:
             query = query.with_for_update()
         if get_all_results:
             results = query.all()
         else:
             results = query.limit(size).offset(offset).all()
 
+        if get_all_results:
+            total = len(results)
+        else:
+            # distinct is required to cope with many-many relationships
+            # (e.g., otherwise you get all 7 results with a total of 460)
+            total = query.order_by(None).distinct(
```

Comment:
adding a distinct is worrying from a performance perspective. Let's check with the benchmarks.

Reply:
baseline:
```
BENCH test_deployments.create: 0:00:53.176378
BENCH test_deployments.list: 0:00:35.887694
```
with the change:
```
BENCH test_deployments.create: 0:00:56.260330
BENCH test_deployments.list: 0:00:36.027018
```
identical to within measurement error, I'd say